mypy/checker.py

```diff
@@ -170,52 +170,54 @@ def check_first_pass(self) -> None:
 
         Deferred functions will be processed by check_second_pass().
         """
-        self.errors.set_file(self.path, self.tree.fullname())
-        with self.enter_partial_types():
-            with self.binder.top_frame_context():
-                for d in self.tree.defs:
-                    self.accept(d)
+        with experiments.strict_optional_set(self.options.strict_optional):
```

Comment:
I'm not sure how likely it is that we ever add more call sites, but I think that's a bug in the making.  The context manager should *always* wrap this code -- I don't think there's any case where we'd want to call this while ignoring the per-file Strict Optional setting.  I understand that you're not a huge fan of code churn, but I don't think it'd be a worthwhile tradeoff in this instance.

Also, FWIW `git blame` has the `-w` flag which ignores whitespace changes, so this doesn't have to mess up anyone's blame.